portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,130 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import java.io.PrintWriter; //to out html
+//Webservlet stuff
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+//for datastore 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+//for querying datastore
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+//for user API
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+//formating json
+import com.google.gson.Gson;
+
+@WebServlet("/loginstat")
+public class LoginServlet extends HttpServlet {
+
+     public final class Login {
+        private boolean loggedIn; //is user logged-in
+        private String email;
+        private String username;
+        private String url;
+
+        public Login(){
+            this.loggedIn = false;
+            this.email = "";
+            this.username = "";
+            this.url = "";
+        }
+        public void setLoggedIn(String userEmail, String userName, String logouturl) {
+            this.loggedIn = true;
+            this.email = userEmail;
+            this.username = userName;
+            this.url = logouturl;
+        }
+        public void setLogoutURL(String logouturl){
+            this.url = logouturl;
+        }
+        /*
+        public Login(boolean logstat, String userEmail, String userName) {
+            this.loggedIn = logstat;
+            this.email = userEmail;
+            this.username = userName;
+        }
+        */
+       
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //set response to json
+        response.setContentType("application/json;");
+
+        UserService userService = UserServiceFactory.getUserService();
+        Login userlogstat = new Login();
+        if (userService.isUserLoggedIn()) {
+            String email = userService.getCurrentUser().getEmail();
+            String username = getUsername(userService.getCurrentUser().getUserId());
+            String logoutUrl = userService.createLogoutURL("/");
+            userlogstat.setLoggedIn(email, username, logoutUrl);
+        }else{
+            String logoutUrl = userService.createLoginURL("/");
```

Comment:
Some comments here to explain why you're calling this `logoutUrl` when you are calling `createLoginURL` will be helpful for readers to understand what's going on